jcstress-samples/src/main/java/org/openjdk/jcstress/samples/high/race/condition/RaceCondition_02_CheckThenReactSequence.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.openjdk.jcstress.samples.high.race.condition;
+
+import org.openjdk.jcstress.annotations.Actor;
+import org.openjdk.jcstress.annotations.JCStressTest;
+import org.openjdk.jcstress.annotations.Outcome;
+import org.openjdk.jcstress.annotations.State;
+import org.openjdk.jcstress.infra.results.II_Result;
+import org.openjdk.jcstress.infra.results.ZZ_Result;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.openjdk.jcstress.annotations.Expect.ACCEPTABLE;
+import static org.openjdk.jcstress.annotations.Expect.FORBIDDEN;
+
+/*
+    How to run this test:
+        $ java -jar jcstress-samples/target/jcstress.jar -t RaceCondition_02_CheckThenReactSequence
+ */
+
+/**
+ * This sample demonstrates you how a check-then-react sequence can lead to surprising results.
+ */
+@JCStressTest
+@Outcome(id = {"true, false", "false, true"}, expect = ACCEPTABLE, desc = "Only one actor got true for the flag in its if-clause")
+@Outcome(id = {"true, true"}, expect = FORBIDDEN, desc = "Both actors got true for the flag in their if-clauses")
```

Comment:
Could you clarify why do you put `forbidden` here?

The implementation does allow `true, true` outcome since it does not synchronize actors.

Reply:
Yes, I can. I considered the outcome "true, true" as the undesirable result of the check-then-react-sequence.
I regard all those race conditions as a kind of "bug patterns" for concurrent programs.
What do you think and what do you propose?

Reply:
If you consider JVM, then the outcome is either allowed by the spec or it is forbidden by the spec. I think the outcomes should be marked from the perspective of the specification.

The outcome true-true is allowed by JVM here, so I expect it to be marked as acceptable or acceptable_interesting.

Reply:
You are right. FORBIDDEN is too much.
Let's choose ACCEPTABLE_INTERESTING.
What do you think?

Reply:
Yes, it should be `ACCEPTABLE_INTERESTING`. That is fully allowed, and it is the target for this test to show such the outcome -- that's the reason for `ACCEPTABLE_INTERESTING` to exist. `FORBIDDEN` means the result is illegal under Java concurrency rules.

Reply:
Ok. Thanks for that explaination. I will keep that in mind from now on.